cmd/kubeadm/app/phases/upgrade/compute.go

```diff
@@ -279,22 +280,27 @@ func GetAvailableUpgrades(versionGetterImpl VersionGetter, experimentalUpgradesA
 }
 
 func getBranchFromVersion(version string) string {
-	return strings.TrimPrefix(version, "v")[:3]
+	var buffer bytes.Buffer
+	v := versionutil.MustParseGeneric(version)
+	buffer.WriteString(fmt.Sprintf("%d", v.Major()))
```

Comment:
You don't need a buffer here. Instead, just do `return fmt.Sprintf("%d.%d", v.Major(), v.Minor())`

Reply:
Yeah, right. That's better. I am some kind of a nut.